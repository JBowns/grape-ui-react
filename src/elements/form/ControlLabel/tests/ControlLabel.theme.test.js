import React from 'react';
import { ControlLabel } from '..';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';

describe('ControlLabel using Theme color', () => {
  it('should return a ControlLabel object with custom color', () => {
    const theme = {
      colors: {
        grapeSoda: {
          base: 'hsl(325, 84.6%, 28%)',
          dark: 'hsl(305, 33.9%, 23.7%)',
          light: 'hsl(313, 67.8%, 47.5%)',
        },
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <ControlLabel bg="grapeSoda.dark" color="grapeSoda.light">Lorem Ipsum</ControlLabel>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('ControlLabel using Theme Font Family', () => {
  it('should return a ControlLabel object with a custom font family', () => {
    const theme = {
      fonts: {
        trueSpace: '"DejaVu Sans Mono", "Bitstream Vera Sans Mono", Courier',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <ControlLabel fontFamily="trueSpace">Lorem Ipsum</ControlLabel>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
