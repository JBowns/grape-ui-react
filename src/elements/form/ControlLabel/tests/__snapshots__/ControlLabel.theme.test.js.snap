// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlLabel using Theme Font Family should return a ControlLabel object with a custom font family 1`] = `
.c0 {
  color: inherit;
  background-color: hsl(0,0%,100%);
  font-family: "DejaVu Sans Mono","Bitstream Vera Sans Mono",Courier;
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  left: 0.5rem;
  padding: 0 0.5rem;
  top: -1px;
  height: 0.4rem;
  position: absolute;
}

<label
  className="c0"
>
  Lorem Ipsum
</label>
`;

exports[`ControlLabel using Theme color should return a ControlLabel object with custom color 1`] = `
.c0 {
  color: hsl(313,67.8%,47.5%);
  background-color: hsl(305,33.9%,23.7%);
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  left: 0.5rem;
  padding: 0 0.5rem;
  top: -1px;
  height: 0.4rem;
  position: absolute;
}

<label
  className="c0"
>
  Lorem Ipsum
</label>
`;
